src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -172,128 +190,93 @@ private KeyWrapProvider CreateKeyWrapProvider(SecurityKey key, string algorithm,
+            if (SupportedAlgorithms.IsSupportedRsaKeyWrap(algorithm, key))
                 return new RsaKeyWrapProvider(key, algorithm, willUnwrap);
 
-            if (key is X509SecurityKey x509Key && SupportedAlgorithms.IsSupportedRsaAlgorithm(algorithm, x509Key))
-                return new RsaKeyWrapProvider(x509Key, algorithm, willUnwrap);
-
-            if (key is JsonWebKey jsonWebKey)
-            {
-                if (jsonWebKey.Kty == JsonWebAlgorithmsKeyTypes.RSA && SupportedAlgorithms.IsSupportedRsaAlgorithm(algorithm, key))
-                {
-                    return new RsaKeyWrapProvider(jsonWebKey, algorithm, willUnwrap);
-                }
-                else if (jsonWebKey.Kty == JsonWebAlgorithmsKeyTypes.Octet && SupportedAlgorithms.IsSupportedSymmetricAlgorithm(algorithm))
-                {
-                    return new SymmetricKeyWrapProvider(jsonWebKey, algorithm);
-                }
-            }
-
-            if (key is SymmetricSecurityKey symmetricKey && SupportedAlgorithms.IsSupportedSymmetricAlgorithm(algorithm))
-                return new SymmetricKeyWrapProvider(symmetricKey, algorithm);
+            if (SupportedAlgorithms.IsSupportedSymmetricKeyWrap(algorithm, key))
+                return new SymmetricKeyWrapProvider(key, algorithm);
 
             throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10661, algorithm, key)));
         }
 
         /// <summary>
-        /// Creates an instance of <see cref="KeyWrapProvider"/> for a specific &lt;SecurityKey, Algorithm>.
+        /// Creates a <see cref="SignatureProvider"/> that creates a signature with the algorithm and <see cref="SecurityKey"/>.
         /// </summary>
-        /// <param name="key">the <see cref="SecurityKey"/> to use.</param>
-        /// <param name="algorithm">the algorithm to use.</param>
-        /// <returns>an instance of <see cref="KeyWrapProvider"/></returns>
-        /// <exception cref="ArgumentNullException">'key' is null.</exception>
-        /// <exception cref="ArgumentNullException">'algorithm' is null or empty.</exception>
-        /// <exception cref="ArgumentException">If <see cref="SecurityKey"/> and algorithm pair are not supported.</exception>
+        /// <param name="key">the <see cref="SecurityKey"/> to use for signing.</param>
+        /// <param name="algorithm">the algorithm to use for signing.</param>
+        /// <exception cref="ArgumentNullException">thrown if <paramref name="key"/> is null.</exception>
+        /// <exception cref="ArgumentNullException">thrown if <paramref name="algorithm"/> is null or empty.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">thrown if <see cref="SecurityKey.KeySize"/> is too small.</exception>
+        /// <exception cref="NotSupportedException">thrown if <see cref="SecurityKey"/> is not assingable from <see cref="AsymmetricSecurityKey"/> or <see cref="SymmetricSecurityKey"/>.</exception>
```

Comment:
nit: assingable -> assignable

Reply:
few more of these in the file

Reply:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/dd4056bb650b8ec7a48d45c0e62e6454da1c7736